File: from-1201-to-1300/1249-minimum-remove-to-make-valid-parentheses/solution.go
func minRemoveToMakeValid(s string) string {
    open := 0
    sl := []byte(s)
    for i := 0; i < len(sl); i ++ {
        if sl[i] == '(' {
            open += 1
        } else if sl[i] == ')' {
            if open == 0 {
                sl[i] = '*'
            } else {
                open -= 1
            }
        }
    }
    for i := len(sl) - 1; i >= 0 && open > 0; i -- {
        if sl[i] == '(' && open > 0 {
            sl[i] = '*'
            open -=1
        }
    }
    result := []string{}
    for i := 0; i < len(sl); i ++ {
        if sl[i] != '*' {
            result = append(result, string(sl[i]))
        }
    }
    return strings.Join(result, "")
}

File: from-1201-to-1300/1249-minimum-remove-to-make-valid-parentheses/solution.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        sl = list(s)
        open = 0
        for i in range(len(sl)):
            if sl[i] == '(':
                open += 1
            elif sl[i] == ')':
                if open == 0:
                    sl[i] = '*'
                else:
                    open -= 1

        i = len(sl) - 1
        while i >= 0 and open > 0:
            if sl[i] == '(' and open > 0:
                sl[i] = '*'
                open -= 1
            i -= 1

        return ''.join([c for c in sl if c != '*'])